anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -483,201 +482,240 @@ def image(dbsession, request_inputs, bodycontent=None):
-                                ),
-                            },
-                        )
+        logger.debug("ADDING/UPDATING IMAGE IN IMAGE POST: " + str(image_info))
 
-                    parent_manifest = json.dumps(image_info.get("parentmanifest", {}))
+        # Check for dockerfile updates to an existing image
+        if (
+            not allow_dockerfile_update
+            and dockerfile
+            and dockerfile_mode.lower() == "actual"
+        ):
+            found_img = db_catalog_image.get(
+                imageDigest=image_info["digest"],
+                userId=account_id,
+                session=dbsession,
+            )
+            if found_img:
+                raise BadRequest(
+                    "Cannot specify dockerfile for an image that already exists unless using force=True for re-analysis",
+                    detail={
+                        "digest": image_info["digest"],
+                        "tag": image_info["fulltag"],
+                    },
+                )
 
-                    logger.debug(
-                        "ADDING/UPDATING IMAGE IN IMAGE POST: " + str(image_info)
-                    )
+        image_records = add_or_update_image(
+            dbsession,
+            account_id,
+            image_info["imageId"],
+            tags=[image_info["fulltag"]],
+            digests=[image_info["fulldigest"]],
+            parentdigest=image_info.get("parentdigest", None),
+            created_at=image_info.get("created_at_override", None),
+            dockerfile=dockerfile,
+            dockerfile_mode=dockerfile_mode,
+            manifest=manifest,
+            parent_manifest=parent_manifest,
+            annotations=annotations,
+        )
+        if image_records:
+            image_record = image_records[0]
 
-                    # Check for dockerfile updates to an existing image
-                    if (
-                        not allow_dockerfile_update
-                        and dockerfile
-                        and dockerfile_mode.lower() == "actual"
-                    ):
-                        found_img = db_catalog_image.get(
-                            imageDigest=image_info["digest"],
-                            userId=userId,
-                            session=dbsession,
-                        )
-                        if found_img:
-                            raise BadRequest(
-                                "Cannot specify dockerfile for an image that already exists unless using force=True for re-analysis",
-                                detail={
-                                    "digest": image_info["digest"],
-                                    "tag": image_info["fulltag"],
-                                },
-                            )
+    except AnchoreApiError:
+        raise
+    except Exception as err:
+        logger.exception("Error adding image")
+        raise err
```

Comment:
The only thing that this block is adding is a log entry mentioning that there is an error adding an image. The cost is high: indents the code into a large try/except block, makes it hard to debug, and it not necessary. Similar as previous comment, for reference, issue: https://github.com/anchore/anchore-engine/issues/867